modules/fs2/src/test/scala/com/snowplowanalytics/snowplow/enrich/fs2/EnrichSpec.scala

```diff
@@ -174,6 +175,99 @@ class EnrichSpec extends Specification with CatsIO with ScalaCheck {
       }
     }
   }
+
+  "resultSink" should {
+    "emit an enriched event with attributes to the good sink" in {
+      TestEnvironment.make(Stream.empty).use { test =>
+        val environment = test.env.copy(goodAttributes = Set("app_id"), piiAttributes = Set(""))
+        val ee = new EnrichedEvent()
+        ee.app_id = "test_app"
+        ee.platform = "web"
+
+        val input = Stream.emit(List(Validated.Valid(ee)))
+
+        for {
+          _ <- input.through(Enrich.resultSink(environment)).compile.drain
+          _ <- test.finalise()
+          good <- test.good.dequeue.compile.toList
+          pii <- test.pii.dequeue.compile.toList
+          bad <- test.bad.dequeue.compile.toList
+        } yield {
+
+          good should beLike {
+            case AttributedData(bytes, attrs) :: Nil =>
+              bytes must not be empty
+              attrs must contain(exactly("app_id" -> "test_app"))
+          }
+
+          (pii should be empty)
+          (bad should be empty)
+        }
+      }
+    }
+
+    "emit a pii event with attributes to the pii sink" in {
+      TestEnvironment.make(Stream.empty).use { test =>
+        val environment = test.env.copy(goodAttributes = Set("app_id"), piiAttributes = Set("platform"))
+        val ee = new EnrichedEvent()
+        ee.app_id = "test_app"
+        ee.platform = "web"
+        ee.pii = "e30="
+
+        val input = Stream.emit(List(Validated.Valid(ee)))
+
+        for {
+          _ <- input.through(Enrich.resultSink(environment)).compile.drain
+          _ <- test.finalise()
+          good <- test.good.dequeue.compile.toList
+          pii <- test.pii.dequeue.compile.toList
+          bad <- test.bad.dequeue.compile.toList
+        } yield {
+
+          good should beLike {
+            case AttributedData(bytes, attrs) :: Nil =>
```

Comment:
This is the error if I change the test so that `good.size > 1`:

```
[error]   x emit a pii event with attributes to the pii sink
[error]    List(AttributedData([B@70e84169,Map(app_id -> test_app)), AttributedData([B@45696b98,Map(app_id -> test_app))) is incorrect:  (EnrichSpec.scala:227)
```

There is no `MatchError` because `beLike` takes a partial function, not a total function.